pkg/kudoctl/cmd/update.go

```diff
@@ -0,0 +1,108 @@
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/kudobuilder/kudo/pkg/kudoctl/cmd/install"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/util/kudo"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+var (
+	updateExample = `
+		The update argument must be a name of the instance.
+
+		# Update dev-flink instance with setting parameter param with value value
+		kubectl kudo upgrade dev-flink -p param=value
+
+		# Update dev-flink instance in namespace services with setting parameter param with value value
+		kubectl kudo upgrade dev-flink -n services -p param=value`
+)
+
+type updateOptions struct {
+	Namespace  string
+	Parameters map[string]string
+}
+
+// defaultOptions initializes the install command options to its defaults
+var defaultUpdateOptions = &updateOptions{
+	Namespace: "default",
+}
+
+// newUpgradeCmd creates the install command for the CLI
+func newUpdateCmd() *cobra.Command {
+	options := defaultUpdateOptions
+	var parameters []string
+	upgradeCmd := &cobra.Command{
+		Use:     "update <instance-name>",
+		Short:   "Update installed KUDO operator.",
+		Long:    `Update installed KUDO operator with new parameters.`,
+		Example: updateExample,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// Prior to command execution we parse and validate passed parameters
+			var err error
+			options.Parameters, err = install.GetParameterMap(parameters)
+			if err != nil {
+				return errors.WithMessage(err, "could not parse parameters")
+			}
+			return runUpdate(args, options)
+		},
+		SilenceUsage: true,
+	}
+
+	upgradeCmd.Flags().StringArrayVarP(&parameters, "parameter", "p", nil, "The parameter name and value separated by '='")
+	upgradeCmd.Flags().StringVar(&options.Namespace, "namespace", defaultOptions.Namespace, "The namespace where the instance you want to upgrade is installed in.")
+
+	const usageFmt = "Usage:\n  %s\n\nFlags:\n%s"
+	upgradeCmd.SetUsageFunc(func(cmd *cobra.Command) error {
+		fmt.Fprintf(upgradeCmd.OutOrStderr(), usageFmt, upgradeCmd.UseLine(), upgradeCmd.Flags().FlagUsages())
+		return nil
+	})
+	return upgradeCmd
+}
+
+func validateUpdateCmd(args []string, options *updateOptions) error {
+	if len(args) != 1 {
+		return errors.New("expecting exactly one argument - name of the instance installed in your cluster")
+	}
+	if len(options.Parameters) == 0 {
+		return errors.New("Need to specify at least one parameter to override via -p otherwise there is nothing to update")
+	}
+
+	return nil
+}
+
+func runUpdate(args []string, options *updateOptions) error {
```

Comment:
We have this patter (which I like) where we separate building of the command from the actual code that is executed in different files. It might seem like overkill here, but these things tend to grow 😉

Reply:
yeah my current guidance is to keep it in one file as long as the command is simple enough...

Reply:
...for now 😉 But I'll leave this up to you